src/lib/operators/abstract_aggregate_operator.cpp

```diff
@@ -0,0 +1,51 @@
+#include "abstract_aggregate_operator.hpp"
+#include "abstract_operator.hpp"
+#include "abstract_read_only_operator.hpp"
+
+#include "types.hpp"
+
+namespace opossum {
+
+AbstractAggregateOperator::AbstractAggregateOperator(const std::shared_ptr<AbstractOperator>& in,
+                                                     const std::vector<AggregateColumnDefinition>& aggregates,
+                                                     const std::vector<ColumnID>& groupby_column_ids)
+    : AbstractReadOnlyOperator(OperatorType::Aggregate, in),
+      _aggregates{aggregates},
+      _groupby_column_ids{groupby_column_ids} {
+  Assert(!(aggregates.empty() && groupby_column_ids.empty()),
```

Comment:
> Btw, the assert will fail, if somebody ever puts a null pointer into the parameters

Hyrise usually does not check for shared_ptrs not being null if the parameter is intended to be non-null only. It would just be too many checks in too many places.